bio-qiime-pipeline/src/utils.py

```diff
@@ -0,0 +1,71 @@
+from typing import Optional, Tuple
+from pathlib import Path
+
+import csv
+
+from coretex import CustomSample, CustomDataset, Experiment
+from coretex.bioinformatics import ctx_qiime2
+
+
+def summarizeSample(sample: CustomSample, outputDir: Path) -> Path:
+    demuxPath = Path(sample.path) / "demux.qza"
+    visualizationPath = outputDir / "demux.qzv"
+
+    ctx_qiime2.demuxSummarize(str(demuxPath), str(visualizationPath))
+    return visualizationPath
+
+
+def determineTruncLen(sample: CustomSample, forward: bool) -> int:
+    sample.unzip()
+    summariesFileName = "forward-seven-number-summaries.tsv" if forward else "reverse-seven-number-summaries.tsv"
+    summariesTsv = list(sample.path.rglob(summariesFileName))[0]
+
+    truncLen: Optional[int] = None
+    with summariesTsv.open("r") as file:
+        summaries = list(csv.reader(file, delimiter = "\t"))
+
+    # summaries will allways have the median quality at row 5 of the csv
+    medianQualitiesStr = summaries[5]
+    medianQualitiesStr.pop(0)  # The first value will be "50%" and not a quality score
+    medianQualities = [float(x) for x in medianQualitiesStr]
+
+    highestScore = 0.0
+    for index, qualityScore in enumerate(medianQualities):
+        qualityScore = float(qualityScore)
```

Comment:
in line 30 you are already converting values of the array to floats and it is being done here again